process/interceptors/factory/eligibleNodesCache.go

```diff
@@ -0,0 +1,102 @@
+package factory
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+type eligibleNodesCache struct {
+	// eligibleNodesMap holds a map with key shard id and value
+	//		a map with key epoch and value
+	//		a map with key eligible node pk and value struct{}
+	// this is needed because:
+	// 	- a shard node will intercept proof for self shard and meta
+	// 	- a meta node will intercept proofs from all shards
+	eligibleNodesMap    map[uint32]map[uint32]map[string]struct{}
```

Comment:
can we use a combined shard+epoch key in the map here?

Reply:
hm.. not sure if this would work in terms of cleanup. when a new epoch is received, with the current implementation it is easy to clean the entries for that shard and other epochs.. using a combined key, that would not be possible